pkg/artifactory/resource_artifactory_local_repository.go

```diff
@@ -256,6 +259,10 @@ func resourceLocalRepositoryDelete(d *schema.ResourceData, m interface{}) error
 
 	resp, err := c.V1.Repositories.DeleteLocal(context.Background(), *repo.Key)
 
+	if resp != nil {
```

Comment:
The equality check should be `resp == nil` but check `err` first. If you check error first you probably never need to check if `resp` is `nil`.